libpod/container_api.go

```diff
@@ -136,6 +137,61 @@ func (c *Container) Update(resources *spec.LinuxResources, restartPolicy *string
 	return c.update(resources, restartPolicy, restartRetries)
 }
 
+// UpdateHealthCheckConfig updates HealthCheck configuration the given container.
+func (c *Container) UpdateHealthCheckConfig(healthCheckConfig *manifest.Schema2HealthConfig, changedTimer bool, noHealthCheck bool) error {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
+	if c.ensureState(define.ContainerStateRemoving) {
+		return fmt.Errorf("container %s is being removed, cannot update: %w", c.ID(), define.ErrCtrStateInvalid)
+	}
+
+	return c.updateHealthCheckConfiguration(healthCheckConfig, changedTimer, noHealthCheck)
+}
+
+// UpdateStartupHealthCheckConfig updates startup HealthCheck configuration the given container.
+func (c *Container) UpdateStartupHealthCheckConfig(startupHealthCheckConfig *define.StartupHealthCheck, changedStartupTimer bool, noHealthCheck bool) error {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
+	if c.ensureState(define.ContainerStateRemoving) {
+		return fmt.Errorf("container %s is being removed, cannot update: %w", c.ID(), define.ErrCtrStateInvalid)
+	}
+
+	return c.updateStartupHealthCheckConfiguration(startupHealthCheckConfig, changedStartupTimer, noHealthCheck)
+}
+
+// UpdateGlobalHealthCheckConfig updates global HealthCheck configuration the given container.
+// If value is nil then value will be not changed.
+func (c *Container) UpdateGlobalHealthCheckConfig(healthLogDestination *string, healthMaxLogCount *uint, healthMaxLogSize *uint, healthCheckOnFailureAction *define.HealthCheckOnFailureAction) error {
```

Comment:
Maybe make a `GlobalHealthCheckOptions` struct that contains all of these? Arguments list is a little long